### PR TITLE
refactor: use console-log-level for logging

### PR DESCRIPTION
**Pull request by ofrobots**
Start using console-log-level so that we can drop the custom logging
code in @google-cloud/common. This also fixes the issue that the logs
were previously getting emitted to stdout instead of stderr.

### CODE REVIEW

File: src/agent/debuglet.ts [hunk 1]
```diff
@@ -255,6 +256,20 @@ export class Debuglet extends EventEmitter {
...
+  //                     0        1        2       3       4        5
+  static LEVELNAMES = [ 'fatal', 'error', 'warn', 'info', 'debug', 'trace'];
+  static logLevelToName(level: number): string {
+    if (typeof level === 'string') {
+      level = parseInt(level, 10);
+    }
+    if (typeof level !== 'number') {
+      level = defaultConfig.logLevel;
+    }
+    if (level < 0) level = 0;
```
Comment (DominicKramer) [inline src/agent/debuglet.ts]: For clarity, I think we should still use `{ ... }` around the body of the if statements.

File: src/agent/debuglet.ts [hunk 2]
```diff
@@ -255,6 +256,20 @@ export class Debuglet extends EventEmitter {
...
+    if (level > 4) level = 4;
```
Comment (DominicKramer) [inline src/agent/debuglet.ts]: Why isn't `else if` used here?

File: src/agent/debuglet.ts (hunk 1 above)
Comment (ofrobots) [reply src/agent/debuglet.ts]: Style guide is okay with this, and I like the improved readability from brevity.

File: src/agent/debuglet.ts (hunk 2 above)
Comment (DominicKramer) [reply src/agent/debuglet.ts]: Ok

**Comment by ofrobots** [comment]
Related: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27506. Will bring that in once it lands.

File: src/agent/debuglet.ts (hunk 1 above)
Comment (DominicKramer) [reply src/agent/debuglet.ts]: I prefer always using brackets, but if the style guide is ok with this then so am I.

**Comment by DominicKramer** [comment]
For record keeping, why did you decide to use `console-log-level`?

**Comment by ofrobots** [comment]
console-log-level serves the purpose we need it for
* It can log to stderr – unlike log-driver.
* It has support for prefixes – something we need.
* It is not a global logger – unlike npmlog – so it is suitable for libraries to use.
* It is small, simple and well-maintained.

File: src/agent/debuglet.ts [hunk 3]
```diff
@@ -14,38 +14,32 @@
  * limitations under the License.
  */
 
-import {AuthenticationConfig, Common} from '../types/common';
-const common: Common = require('@google-cloud/common');
-
+import * as assert from 'assert';
```
Comment (JustinBeckwith) [inline src/agent/debuglet.ts]: Why import assert?

File: src/agent/debuglet.ts [hunk 4]
```diff
@@ -14,38 +14,32 @@
...
 import * as debugapi from './v8/debugapi';
-
-import * as assert from 'assert';
-
-import * as stackdriver from '../types/stackdriver';
-import {DebugAgentConfig, ResolvedDebugAgentConfig} from './config';
-import {Debug, PackageInfo} from '../client/stackdriver/debug';
-import {Logger} from '../types/common';
 import {DebugApi} from './v8/debugapi';
+import consoleLogLevel = require('console-log-level');
```
Comment (JustinBeckwith) [inline src/agent/debuglet.ts]: It looks like there is an @types/console-log-level package, can we use that?

Comment (ofrobots) [reply src/agent/debuglet.ts]: Absolutely! That's why I wrote it.

File: src/agent/debuglet.ts (hunk 3 above)
Comment (ofrobots) [reply src/agent/debuglet.ts]: We have always imported `assert` – there are some debugging assertions in the code. `gts fix` reorganized the imports for whatever reason.

**Pull request merged**